### ISSUE TITLE
Missing pagination information in response

### ISSUE THREAD

**Issue by subashsn** [jdalrymple/node-gitlab#94]
When I use the `Projects.all()` call to list projects it by default lists up to 30 projects. But if there are more than `perPage` number of projects, there is no way to know based on the data returned by the function. This makes it difficult to implement pagination in UI.

The Gitlab API returns this information as part of the headers

Example
```
X-Next-Page: 3
X-Page: 2
X-Per-Page: 3
X-Prev-Page: 1
X-Total: 8
X-Total-Pages: 3
```

Am I missing something? What is the best way to get the pagination information along with the response? 

Thank you :)

**Comment by jdalrymple** [comment]
The library should return all the projects, as the pagination is handled in the background. We changed it in the past to not just return 30 results since it was misleading. 30 != all lol. If it is only returning 30, then something is probably broken :( I can look into this?

**Comment by jdalrymple** [comment]
So i just tested this, it still returns all the results which is good. Would you like more information when using the paging options?

**Comment by subashsn** [comment]
Hey, thanks for the reponse!

I am using the api using
```

**Comment by jdalrymple** [comment]
Very doable. I'll add in an option to also retrieve the pagination information

**Comment by jdalrymple** [comment]
In this example,

```
var client = new Gitlab({url: process.env.GITLAB_URL, oauthToken: accessToken})
client.Projects.all({page:page,owned:true}).then(function(projects){
...
```

page is variable right? You can also get more results if you supply the perPage option which will give you more results per page.

**Comment by subashsn** [comment]
Yep. `page`, `perPage` options are passed to GitLab and I am able to control the page and number of results per page.

**Comment by jdalrymple** [comment]
So ive been thinking about the cleanest way to do this but havent come up with the best solution :s. pretty much just an expanded response:

```javascript
{
   data: []
   pagination: {
   }
}
```

Thoughts?

**Comment by subashsn** [comment]
I think that should work!

So I can access the pagination data by doing something like this?
```js
client.Projects.all({page:page,owned:true}).then((projects, pagination) => {
    var totalPages = pagination['X-Total']
```

It doesnt't break anyones code either. So I think this is a pretty good way to add this.

Thanks :)

**Comment by jdalrymple** [comment]
```
client.Projects.all({ page:page, owned:true, pagination: true})
.then((projects, pagination) => {
    var totalPages = pagination['X-Total']
```

Should I abstract out the pagination keys too?

Like:
```
pagination: {
  next: 4,
  page: 2,
  perPage: 3,
  prev: 1,
  total: 3,
```
OR
```
pagination: {
  nextPage: 4,
  page: 2,
  perPage: 3,
  prevPage: 1,
  totaPagesl: 3,
```

**Comment by subashsn** [comment]
Yep, that would make it really clean!

```js
pagination: {
  nextPage: 4,
  currentPage: 2,
  itemsPerPage: 3,
  previousPage: 1,
  totalPages: 3
}
```
Maybe too verbose :P But anything works, really.

Thanks :D

**Comment by jdalrymple** [comment]
Just added the functionality to the pagination branch, using this syntax: 

```
pagination: {
  next: 4,
  current: 2,
  perPage: 3,
  previous: 1,
  total: 3,
}
```

**Comment by jdalrymple** [comment]
just published! Tell me if it works for you

**Comment by subashsn** [comment]
I was not able to get this working.

This is what I tried
```js
    client.Projects.all({page:page,owned:true,pagination:true}).then((projects,pagination) => {
        console.log(pagination)
```

But the `pagination` variable is still printing `undefined` :(

Thanks

**Comment by jdalrymple** [comment]
Why projects.paginatiom ?

**Comment by jdalrymple** [comment]
The return would be ({ data, pagination}) => { console.log(pagination)}

**Comment by jdalrymple** [comment]
And the option is showPagination : true. Changed it for clarity

**Comment by jdalrymple** [comment]
Perks of coding while tired smh. I'll fix it tonight sorry

**Comment by subashsn** [comment]
No problem! And thanks a lot :)

**Comment by jdalrymple** [comment]
Fixing it now, and adding tests so i dont make the same mistake in the future :P

**Comment by jdalrymple** [comment]
Will release today! Added tests as well.

**Comment by jdalrymple** [comment]
Oh and also note the properties have changed a bit. Check the readme :)
